signup-service/src/main/java/io/mosip/signup/services/IdentityVerificationService.java

```diff
@@ -147,27 +164,38 @@ public SlotResponse getSlot(String transactionId, SlotRequest slotRequest) {
 
     private String fetchAndVerifyAccessToken(String authCode) {
         try {
-            AuthorizationCode code = new AuthorizationCode(authCode);
-            URI callback = new URI(oauthRedirectUri);
-            AuthorizationGrant codeGrant = new AuthorizationCodeGrant(code, callback);
-
             long issuedTime = System.currentTimeMillis();
             JWTClaimsSet.Builder builder = new JWTClaimsSet.Builder()
                     .subject(oauthClientId)
                     .issuer(oauthClientId)
-                    .audience(oauthIssuerUri)
+                    .audience(oauthIssuerUri+"/v1/esignet/oauth/token")
                     .issueTime(new Date(issuedTime))
                     .expirationTime(new Date(issuedTime+(60*1000)));
             JWSHeader jwsHeader = new JWSHeader(JWSAlgorithm.RS256);
             SignedJWT signedJWT = new SignedJWT(jwsHeader, builder.build());
-            //signedJWT.sign(new RSASSASigner()); TODO Read private key from keystore
-            //ClientAuthentication clientAuthentication = new PrivateKeyJWT(signedJWT);
+            PrivateKey privateKey = loadPrivateKey(privateKeyAlias, p12FilePassword);
+            signedJWT.sign(new RSASSASigner((RSAPrivateKey) privateKey));
+
+            // Create a Token Request with the authorization code
+            AuthorizationCode code = new AuthorizationCode(authCode);
+            URI callback = new URI(oauthRedirectUri);
+            URI tokenEndpoint = new URI(oauthIssuerUri+"/v1/esignet/oauth/token");
+            AuthorizationGrant codeGrant = new AuthorizationCodeGrant(code, callback);
+            ClientAuthentication clientAuthentication = new PrivateKeyJWT(signedJWT);
+            TokenRequest request = new TokenRequest(tokenEndpoint,clientAuthentication, codeGrant);
+            HTTPRequest toHTTPRequest = request.toHTTPRequest();
+            TokenResponse tokenResponse= OIDCTokenResponseParser.parse(toHTTPRequest.send());
+            OIDCTokenResponse successResponse = (OIDCTokenResponse) tokenResponse.toSuccessResponse();
```

Comment:
What if the token endpoint fails? should we consider checking the failedResponse before checking the successResponse?

Reply:
added error respose